From 53904e21adc18214b426c33673190b7ef4ed340a Mon Sep 17 00:00:00 2001
From: Stephen Cavilia <sac+nongnu@atomicradi.us>
Date: Fri, 22 Jun 2012 10:59:05 -0400
Subject: [PATCH] Accept Freescale UARTS as RS232 devices.

The on-chip UARTs on Freescale i.MX processors use device names
/dev/ttymxc[012] and major number 207 on linux. These will get classified as
an unknown source type by gpsd. Attached patch makes them classify as standard
RS232 devices.
---
 serial.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/serial.c
+++ b/serial.c
@@ -54,7 +54,8 @@ static sourcetype_t gpsd_classify(const
 	 * ftp://ftp.kernel.org/pub/linux/docs/device-list/devices-2.6+.txt
 	 */
 	int devmajor = major(sb.st_rdev);
-	if (devmajor == 4 || devmajor == 204)
+        /* 207 are Freescale i.MX UARTs (ttymxc*) */
+	if (devmajor == 4 || devmajor == 204 || devmajor == 207)
 	    devtype = source_rs232;
 	else if (devmajor == 188)
 	    devtype = source_usb;
